New article - Create Azure functions for SharePoint webhooks using an azd template

Yvand opened the pull request:
## Category

- [ ] Content fix
- [x] New article

## What's in this Pull Request?

Proposing a new article to create Azure functions for SharePoint webhooks using an azd template

Yvand commented:
@andrewconnell thank you for your review.
I am not sure to understand your question about the TOC, but this article provides an alternative to https://learn.microsoft.com/en-us/sharepoint/dev/apis/webhooks/sharepoint-webhooks-using-azure-functions (it achieves exactly the same goal in a different way)

VesaJuvonen commented:
Looks like a great new doc with good content which we want to get updated. @andrewconnell will clean up things a bit to align with the learn publishing practices and we'll get it then merged and published after that's completed.

VesaJuvonen reviewed (approved):
Good to go after addressing the formatting changes.

andrewconnell commented on docs/apis/webhooks/sharepoint-webhooks-using-azd-template.md:
```diff
@@ -0,0 +1,178 @@
+---
+title: Create Azure functions for SharePoint webhooks using an azd template
```
Azure Functions should be sentence case, not lcase `f`

andrewconnell commented on docs/apis/webhooks/sharepoint-webhooks-using-azd-template.md:
```diff
@@ -0,0 +1,178 @@
+# Azure functions for SharePoint webhooks using azd
+
+[Azure Developer CLI (azd)](https://aka.ms/azd) is an open-source tool that accelerates provisioning and deploying app resources in Azure.  
+This article uses the [Azure function app for SharePoint webhooks public template](https://github.com/Azure-Samples/azd-functions-sharepoint-webhooks) to deploy an Azure function app that connects to your SharePoint Online tenant, to register and manage [webhooks](overview-sharepoint-webhooks.md), and process the notifications from SharePoint.
```
Paragraphs should be separated by newlines, not adjacent like this

ref: MSDOCS & MSLEAN content & markdown formatting rules

andrewconnell commented on docs/apis/webhooks/sharepoint-webhooks-using-azd-template.md:
```diff
@@ -0,0 +1,178 @@
+- [Node.js 20](https://www.nodejs.org/)
+- [Azure Functions Core Tools](/azure/azure-functions/functions-run-local?pivots=programming-language-typescript#install-the-azure-functions-core-tools)
+- [Azure Developer CLI (azd)](/azure/developer/azure-developer-cli/install-azd)
+- An Azure subscription trusting the same Entra ID directory as your SharePoint tenant
```
- "Entra ID" should be "Microsoft Entra ID" for the first reference
- "trusting" should be "that trusts"
- avoid 2nd person ... favor 3rd person... in this case "your" should be "the"

andrewconnell commented on docs/apis/webhooks/sharepoint-webhooks-using-azd-template.md:
```diff
@@ -0,0 +1,178 @@
+## Prerequisites
+
+- [Node.js 20](https://www.nodejs.org/)
+- [Azure Functions Core Tools](/azure/azure-functions/functions-run-local?pivots=programming-language-typescript#install-the-azure-functions-core-tools)
```
Remove the deep link unless it's absolutely necessary. In this case:
- the install hash isn't necessary
- the language-specific `pivot` shouldn't be included

ref: MSDOCS & MSLEAN content & markdown formatting rules

andrewconnell commented:
@Yvand I didn't intend to send single reviews... my mistake... still working on the doc today and will send the rest in batch. Thanks for taking care of those others promptly though

andrewconnell commented on docs/apis/webhooks/sharepoint-webhooks-using-azd-template.md:
```diff
@@ -0,0 +1,179 @@
+
+`DefaultAzureCredential` will use a managed identity to authenticate to SharePoint. This may be the existing, system-assigned managed identity of the function app service or a user-assigned managed identity.  
+
+This tutorial will assume that the system-assigned managed identity is used.
```
MSDOCS guidelines are to be less wordy and more direct/concise, so... 

"will assume that the" > "assumes the"

andrewconnell commented on docs/apis/webhooks/sharepoint-webhooks-using-azd-template.md:
```diff
@@ -0,0 +1,179 @@
+
+#### Grant the SharePoint API permission Sites.Selected to the managed identity
+
+Navigate to your function app in [the Azure portal](https://portal.azure.com/#blade/HubsExtension/BrowseResourceBlade/resourceType/Microsoft.Web%2Fsites/kind/functionapp) > click `Identity` and note the `Object (principal) ID` of the system-assigned managed identity.  
```
"the" shouldn't be linked for the portal.

andrewconnell commented on docs/apis/webhooks/sharepoint-webhooks-using-azd-template.md:
````diff
@@ -0,0 +1,179 @@
+}
+New-MgServicePrincipalAppRoleAssignment -ServicePrincipalId $managedIdentityObjectId -BodyParameter $appRoleAssignment | Format-List
+```
+
````
If this extra newline isn't necessary, remove it so it doesn't create additional space.... noticed this for the other `<details>` blocks.

andrewconnell commented on docs/apis/webhooks/sharepoint-webhooks-using-azd-template.md:
```diff
@@ -0,0 +1,179 @@
+Then, use one of the scripts below to grant this identity the app-only permission `Sites.Selected` on the SharePoint API:
+
+> [!IMPORTANT]
+> The scripts below require at least the delegated permission [`AppRoleAssignment.ReadWrite.All`](/graph/permissions-reference#approleassignmentreadwriteall) (requires admin consent)
```
this line & the ones above it that reference GUIDs & scopes should be bold formatting, not inline code

andrewconnell replied:
Noticed this present in other places in the doc that should also be addressed

Yvand commented:
@andrewconnell thank you for your time to review it.
I hope I correctly addressed all your points, my only doubt is about DefaultAzureCredential:
Not sure if it should be `DefaultAzureCredential` or **DefaultAzureCredential**, I kept `DefaultAzureCredential`, but I will change it if needed.

andrewconnell reviewed (changes requested):
All looks good, but maybe one more edit for discoverability... [add a callout maybe in this doc](https://github.com/SharePoint/sp-dev-docs/blob/main/docs/apis/webhooks/sharepoint-webhooks-using-azure-functions.md)?

Yvand commented:
This is a great suggestion, I updated https://learn.microsoft.com/sharepoint/dev/apis/webhooks/sharepoint-webhooks-using-azure-functions to add a reference to this new article, I hope the change is fine.

jansenbe commented:
@Yvand : Really nice work, thanks for this contribution!

@andrewconnell : I'm OK with merging this one
